docs/coding-guidelines/clr-code-guide.md

```diff
@@ -1252,10 +1256,30 @@ This data is meant to be changed in a scoped manner only. In particular, the CON
 
 See the big block comment at the start of [src\inc\contract.h][contract.h].
 
-## <a name="2.11"></a>2.11 Is your code DAC compliant?
+## <a name="2.11"></a>2.11 Using standard headers
+
+The C and C++ standard headers are available for usage in the CoreCLR code-base. However, there are a few restrictions on using the standard-provided APIs for code that will run as part of CoreCLR.
+
+Code that will only run in other processes, such as createdump or other extraneous tools, do not have many of these restrictions.
+
+### <a name="2.11.1"></a> 2.11.1 Do not use wchar_t
+
+The `wchar_t` type is implementation-defined, with Windows and Unix-based platforms using different definitions (2 byte vs 4 byte). Use the `WCHAR` alias instead, which is always 2 bytes. The CoreCLR PAL provides implementations of a variety of the C standard `wchar_t` APIs with the `WCHAR` type instead. These methods, as well as the methods in the CoreCLR minipal and in the repo minipal should be used. If a CoreCLR minipal API exists, it should be used instead of the PAL API.
```

Comment:
Perhaps add a link to the top level directory for the PAL and minipal?

Reply:
Should we mention `unsigned char` used by non-win32 in minipal/utf8 internals and usages: https://github.com/dotnet/runtime/blob/6fc9636509d014af530add71a6186218be90df55/src/native/minipal/utf8.h#L21-L25

Reply:
Should we mention `unsigned short` used by non-win32 in minipal/utf8 internals and usages: https://github.com/dotnet/runtime/blob/6fc9636509d014af530add71a6186218be90df55/src/native/minipal/utf8.h#L21-L25